libvmi/driver/xen/libxc_wrapper.c

```diff
@@ -97,8 +97,20 @@ status_t create_libxc_wrapper(xen_instance_t *xen)
 
     if ( !wrapper->handle )
     {
-        fprintf(stderr, "Failed to find a suitable libxenctrl.so at any of the standard paths!\n");
-        return VMI_FAILURE;
+        char *alternate = g_malloc0(snprintf(NULL, 0, "libxenctrl.so.%u.%u",
+                                             xen->major_version, xen->minor_version)+1);
+        sprintf(alternate, "libxenctrl.so.%u.%u", xen->major_version, xen->minor_version);
+
+        dbprint(VMI_DEBUG_XEN, "--libxc_wrapper looking for %s\n", alternate);
+
+        wrapper->handle = dlopen (alternate, RTLD_NOW | RTLD_GLOBAL);
+        free(alternate);
```

Comment:
Not sure if its still the case but if you have certain options on you will
have issues. I think this may also introduce issues for any hope of windows
compat due to the memory pools it pulls from when malloc is called.

On Tue, Aug 23, 2016 at 3:30 PM, Tamas K Lengyel <notifications@github.com>
wrote:

> In libvmi/driver/xen/libxc_wrapper.c
> <https://github.com/libvmi/libvmi/pull/368#discussion_r75934705>:
>
> > @@ -97,8 +97,20 @@ status_t create_libxc_wrapper(xen_instance_t *xen)
> >
> >      if ( !wrapper->handle )
> >      {
> > -        fprintf(stderr, "Failed to find a suitable libxenctrl.so at any of the standard paths!\n");
> > -        return VMI_FAILURE;
> > +        char *alternate = g_malloc0(snprintf(NULL, 0, "libxenctrl.so.%u.%u",
> > +                                             xen->major_version, xen->minor_version)+1);
> > +        sprintf(alternate, "libxenctrl.so.%u.%u", xen->major_version, xen->minor_version);
> > +
> > +        dbprint(VMI_DEBUG_XEN, "--libxc_wrapper looking for %s\n", alternate);
> > +
> > +        wrapper->handle = dlopen (alternate, RTLD_NOW | RTLD_GLOBAL);
> > +        free(alternate);
>
> There is no difference between free and g_free AFAICT.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/libvmi/libvmi/pull/368/files/367d3c335a62514d8d160de672b7027949e02db9#r75934705>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADBrK4-PYayC0ZnwjbOzLOsEv_bK5iSHks5qi0pCgaJpZM4JrEEz>
> .
>



-- 
Matthew Fusaro
Partner & Security Engineer
Zentific
*Secure More, Worry Less*